typesetters/base.lua

```diff
@@ -6,6 +6,40 @@
 
 -- Typesetter base class
 
+SILE.nodefactory.discretionary.outputYourself = function (self, typesetter, line)
```

Comment:
Forgot to move this at due place and explain it.

Reply:
(and it can even be made a bit less naive)

Reply:
See https://github.com/sile-typesetter/sile/pull/1980
Let's discuss that first, and I refactor/rebase this very PR here afterwards.

Reply:
Fixed = Rebased on develop